Add documentation about configuring images in Harbor to be private (#291)
alphagov/gsp: docs/gds-supported-platform/making-things-private-in-harbor.md

Making images private in Harbor
===============================

Sometimes, we need to make our Docker images private. By default, the GSP Harbor
is publicly accessible. Here are instructions on how to make images
private.

1. Add the following to your build-pipeline.yml:

```
  harbor_source: &harbor_source
    username: ((harbor.harbor_username))
    password: ((harbor.harbor_password))
    harbor:
      url: ((harbor.harbor_url))
      prevent_vul: "false"
      public: "false"
    notary:
      url: ((harbor.notary_url))
      root_key: ((harbor.root_key))
      delegate_key: ((harbor.ci_key))
      passphrase:
        root: ((harbor.notary_root_passphrase))
        snapshot: ((harbor.notary_snapshot_passphrase))
        targets: ((harbor.notary_targets_passphrase))
        delegation: ((harbor.notary_delegation_passphrase))

  resource_types:
  - name: harbor
    type: docker-image
    privileged: true
    source:
      repository: govsvc/gsp-harbor-docker-image-resource
      tag: "0.0.1553882420"

  resources:
  - name: my-image
    type: harbor
    source:
      <<: *harbor_source
      repository: registry.((cluster.domain))/my-collection/my-image
```

> The crucial line is `public: false` to ensure that anonymous users (ie

Reviewer: ```suggestion
The crucial line is `public: "false"` to ensure that anonymous users (ie
```
The content here should match the content above

Reviewer 2: Good shout. This boolean NEEDS to be a string, otherwise the API won't accept it.

Author: It does now. :flushed:

Author: We found this out the hard way, IIRC.

public users) can't pull images from that Harbor repo.

2. Add `imagePullSecrets` to your Helm chart in the section where you

>    define pods and their containers:

Reviewer: is it worth mentioning `registry-creds` by name up here? Otherwise it's not clear what exactly is happening.  Something like

```
2. Add `imagePullSecrets` to your Helm chart in order to use the builtin secret `registry-creds` for pulling the private docker image:
```

Author: Thanks, I've used that entire line.

Making images private in Harbor
===============================

> DCS has a need for private Docker images. By default, the GSP Harbor

Reviewer 3: Maybe don't need to call out DCS in the generic docs

Author: ```suggestion
Sometimes, we need to make our Docker images private. By default, the GSP Harbor
```
